includes/wp-admin/class-admin.php

```diff
@@ -593,23 +598,207 @@ public static function user_bulk_options( $actions ) {
+		}
+
+		// Check permissions.
+		if ( ! \current_user_can( 'edit_users' ) ) {
+			\wp_die( \esc_html__( 'You do not have sufficient permissions to perform this action.', 'activitypub' ) );
+		}
+
+		// Get form data.
+		// phpcs:ignore WordPress.Security.ValidatedSanitizedInput
+		$selected_users = \wp_unslash( $_POST['selected_users'] ?? array() );
+		// phpcs:ignore WordPress.Security.ValidatedSanitizedInput
+		$remove_from_fediverse = \wp_unslash( $_POST['remove_from_fediverse'] ?? array() );
+		$send_back             = \esc_url_raw( \wp_unslash( $_POST['send_back'] ?? '' ) );
+
+		// Sanitize user IDs.
+		$selected_users = \array_map( 'absint', (array) $selected_users );
+		$selected_users = \array_filter( $selected_users );
+
+		if ( empty( $selected_users ) ) {
+			\wp_safe_redirect( $send_back );
+			exit;
+		}
+
+		// Process capability removal using unified method.
+		$result = self::process_capability_removal( $selected_users, $remove_from_fediverse, $send_back );
+
+		// Redirect back.
+		\wp_safe_redirect( $result );
+		exit;
+	}
+
+
+	/**
+	 * Process fediverse deletion for users (capabilities already removed).
+	 *
+	 * @param array        $users                  Array of user IDs.
+	 * @param array|string $remove_from_fediverse  Array of user IDs to delete from fediverse, or 'delete'/'keep' for all users.
+	 * @param string       $send_back              URL to redirect back to.
+	 *
+	 * @return string The URL to redirect to.
+	 */
+	public static function process_capability_removal( $users, $remove_from_fediverse, $send_back ) {
+		$deleted_count = 0;
+
+		// Normalize fediverse removal parameter.
+		if ( is_string( $remove_from_fediverse ) ) {
+			// Legacy format: 'delete' or 'keep' for all users.
+			$delete_all      = ( 'delete' === $remove_from_fediverse );
+			$users_to_delete = $delete_all ? array_flip( $users ) : array();
+		} else {
+			// New format: array of specific user IDs to delete from fediverse.
+			$remove_from_fediverse = \array_map( 'absint', (array) $remove_from_fediverse );
+			$users_to_delete       = \array_flip( \array_filter( $remove_from_fediverse ) );
+		}
+
+		// Schedule delete activities for users who should be removed from fediverse.
+		if ( ! empty( $users_to_delete ) ) {
+			// Temporarily bypass capability checks for delete activity scheduling since capabilities were already removed.
+			\add_filter( 'activitypub_user_can_activitypub', '__return_true' );
+
+			foreach ( $users as $user_id ) {
+				if ( isset( $users_to_delete[ $user_id ] ) ) {
+					Actor::schedule_user_delete( $user_id );
+					++$deleted_count;
+				}
+			}
+
+			\remove_filter( 'activitypub_user_can_activitypub', '__return_true' );
+		}
+
+		// Add success message parameters to the redirect URL.
+		$message_params = array();
+
+		if ( $deleted_count > 0 ) {
+			$message_params['activitypub_bulk_deleted'] = $deleted_count;
+		}
+
+		return \add_query_arg( $message_params, $send_back );
```

Comment:
```suggestion
		if ( $deleted_count ) {
			$send_back = \add_query_arg( 'activitypub_bulk_deleted', $deleted_count, $send_back );
		}

		return $send_back;
```
Simplifies it a bit.

It doesn't look like we do anything with `activitypub_bulk_deleted`? I didn't get an admin notice after removing caps from multiple users.